src/redux/reducers/TodoReducer.ts

```diff
@@ -0,0 +1,77 @@
+import { TodoItemType } from '../../models';
+import { AddItemPayload } from '../actions/todoActions/AddItemAction';
+import { ItemDeletePayload } from '../actions/todoActions/deleteItemAction';
+import { TodoActionTypes } from '../actions/todoActions';
+import { DoneItemPayload } from '../actions/todoActions/doneItemAction';
+import { SignInPayload } from '../actions/todoActions/signInAction';
+
+export type TodoReducerState = {
+  todoItems: TodoItemType[];
+  userToken: SignInPayload['userToken'] | null;
+  user: SignInPayload['user'] | null;
+};
+
+const initialState: TodoReducerState = {
+  todoItems: [],
+  userToken: null,
+  user: null,
+};
+
+export const todoReducer = (
+  state = initialState,
+  action: any,
+): TodoReducerState => {
+  switch (action.type) {
+    case TodoActionTypes.ADD_ITEM: {
+      const { newItem }: AddItemPayload = action.payload;
+
+      return {
+        ...state,
+        todoItems: [...state.todoItems, newItem],
+      };
+    }
+    case TodoActionTypes.DELETE_ITEM: {
+      const { id }: ItemDeletePayload = action.payload;
+
+      const newTodoItems = [
+        ...state.todoItems.filter(item => {
```

Comment:
we already discussed it, why did you skip it?

Reply:
sory, I will fix it